[ChrisHo1341] iP

ChrisHo1341 opened the pull request:
# Meet Chatimous Maximous! Your posh personal assistant!

> "Never be the smartest person in the room." - The Big Diesel

### Chatimous Maximous will keep your mind at ease and help you through a busy day! He can:
- Tell you what tasks you need to complete
- Tell you when you need to complete them
- Tell you whether or not you have completed them

### When would he be useful to you?
1. When you are a busy person who needs to keep track of many things.
2. When you don't have a personal secretary.
3. When you ~~like~~ LOVE to keep organized!

If you fall under any of those categories, Chatimous Maximous is the text-assistant for you!
All you need to do is download from [here](https://github.com/ChrisHo1341/ip)!

Here is a sneak peak of how we keep track of a `deadline` :upside_down_face::
```
    /**
     * Constructor for new deadlines.
     * @param name Description or name of the given task.
     * @param date Date that the task needs to be completed by.
     */
    public Deadline(String name, LocalDate date) {
        super(name);
        this.date = date;
    }
```
### Planned features:
- [x] Keep track and tell you the tasks you need to do
- [x] Keep track of important dates
- [ ] Custom replies based using GenAI
- [ ] Android body to help you with chores

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ChrisHo2339 committed:
Level-0

Level-0 thing

ChrisHo2339 committed:
Level 1

ChrisHo2339 committed:
Update Duke.java and add TaskList

Add new feature to Duke.
Store all inputs into a list and output the full list when "list" is input

ChrisHo2339 committed:
Update Duke.java and add Task class

Add marking feature for tasks

ChrisHo2339 committed:
Update Duke.java and add different types of tasks

Add new types of tasks such as todo, event and deadline. Create chatbot class.

ziyang27 commented on src/main/java/ChatBot.java:
```diff
@@ -0,0 +1,64 @@
+public class ChatBot {
+    private static final String INDENT = "    ";
+    private static final String NEW_LINE = INDENT + "____________________________________________________________ \n";
+    private static final String GREETING = INDENT + "Good day good sir! I am Chatimous Maximous, here to help you with your every need!\n";
+    private static final String BYE = "bye";
+    private static final String LIST = "list";
+    private static final String UNMARK = "unmark";
+    private static final String MARK = "mark";
+    private static final String EXIT = INDENT + "It has been a pleasure! I do hope to see you again! \n";
+    private static final String TODO = "todo";
+    private static final String DEADLINE = "deadline";
+    private static final String PRAISE = NEW_LINE + "A job well done Sir! \n" + NEW_LINE;
+    private static final int SEVEN = 7;
+    private static final String ENCOURAGEMENT = NEW_LINE + "Don't fret Sir! You'll get it soon \n" + NEW_LINE;
+    private static final int FIVE = 5;
+    private static final String EVENT = "event";
+    private static final String ADD_TASK = " has been accounted for! \n" + NEW_LINE;
+    private static final String NORMAL = "normal";
```
I like how you used uppercase for all the constant names 👍

ChrisHo1341 replied:
Thank you for the help!

ziyang27 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,36 @@
+        this.taskType = task.charAt(0);
+    }
+
+    public void doneTask() {
```
Perhaps you can change the method name to markTask which I feel would be a little clearer? 🤔

ziyang27 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,36 @@
+        }
+    }
+
+    public void completeTask() {
```
Hmm, I think there is a repeated method here (with doneTask). Maybe can consider combining them into one to shorten the code?

ziyang27 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    public String showList() {
+        Task task;
+        String status;
+        String finalOutput = new String("");
+        
+        for (int i = 0; i < taskList.size(); i++) {
+            task = taskList.get(i);
+            status = task.checkStatus();
+            finalOutput = finalOutput + indent + Integer.toString(i + 1) + "." + status + "\n";
+        }
+        return finalOutput;
```
String concatenation can be quite expensive, maybe you can consider trying using StringBuilder? Other than that clean and readable code 👍

ziyang27 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,30 @@
+        ChatBot Chatty = new ChatBot();
+        Scanner reader = new Scanner(System.in);
+        String input = reader.nextLine();
+        String message;
```
Hi, small issue but the coding standard, under the 'Variables' section states that variables should be initialized where they are declared 😢 Personally, I also make the same mistake but just take note in the future 👍

ziyang27 reviewed (commented):
LGTM. The code is really clean and readable 👍 There's just some minor improvements to consider. Keep it up 😄

ChrisHo2339 committed:
Update ChatBot.java

Add exception handling for invalid inputs

ChrisHo2339 committed:
Update TaskList.java and ChatBot.java

Add feature to delete tasks from task list

ChrisHo2339 committed:
Update chatbot

Change some outputs

ChrisHo2339 committed:
Add automatic testing

Add automatic testing

ChrisHo1341 committed:
Merge pull request #1 from ChrisHo1341/branch-level-7

Update chatbot

ChrisHo2339 committed:
Add save and load

Add saving and loading functionality for chatbot to save when the task list changes and load upon start up

ChrisHo1341 committed:
Merge pull request #3 from ChrisHo1341/branch-level-7

Add save and load

ChrisHo2339 committed:
Add date feature

Add feature that recognizes date formats for deadlines and events

ChrisHo1341 committed:
Merge pull request #4 from ChrisHo1341/branch-level-8

Add date feature

ChrisHo2339 committed:
Add Ui class

Add Ui class

ChrisHo1341 committed:
Merge pull request #6 from ChrisHo1341/branch-A-MoreOOP

Add Ui class

ChrisHo2339 committed:
Package classes

Package classes

ChrisHo1341 committed:
Merge pull request #7 from ChrisHo1341/branch-A-package

Package classes

ChrisHo1341 committed:
Merge pull request #8 from ChrisHo1341/add-gradle-support

Add gradle support

ChrisHo2339 committed:
Add gradle support

Add gradle support

ChrisHo2339 committed:
Add JUnit tests

Add JUnit tests for Storage method and TaskList method

ChrisHo2339 committed:
Create settings.json

ChrisHo2339 committed:
Add JavaDoc

Add JavaDoc for each class and method

ChrisHo2339 committed:
Update code

Update code according to coding standard

ChrisHo2339 committed:
Add find functionality

Add find command to allow users to search for tasks

ChrisHo1341 committed:
Merge pull request #9 from ChrisHo1341/branch-level-9

Add find functionality

ChrisHo1341 committed:
Merge pull request #10 from ChrisHo1341/branch-A-JavaDoc

Add JavaDoc

ChrisHo1341 committed:
Merge branch 'master' into branch-A-CodingStandard

ChrisHo1341 committed:
Merge pull request #11 from ChrisHo1341/branch-A-CodingStandard

Update code

ChrisHo2339 committed:
Add checkstyle

Add Checkstyle to detect coding style violations

Will amend found violations later at later time

ChrisHo1341 committed:
Merge pull request #12 from ChrisHo1341/branch-A-Checkstyle

Add checkstyle

ChrisHo2339 committed:
Add GUI

Add GUI to chatbot

ChrisHo1341 committed:
Merge pull request #13 from ChrisHo1341/branch-level-10

Add GUI

ChrisHo2339 committed:
Update GUI

Update Chatbot to use responses from ChatBot()

Previous iteration only featured a working GUI template lacking the backend chatbot.

This commit adds the backend functionality to the GUI to give an interactive experience for users.

ChrisHo1341 committed:
Merge pull request #14 from ChrisHo1341/branch-level-10

Update GUI

ChrisHo2339 committed:
Update TaskList.java

Add constructor to TaskList.java to work with GUI

Previous iteration ran into bug where TaskList would double count tasks from txt file.

Bug was successfully fixed by adding a constructor the TaskList class.

ChrisHo1341 committed:
Merge pull request #15 from ChrisHo1341/branch-level-10

Update TaskList.java

ChrisHo2339 committed:
Update code

Update all code to follow Java Coding Standards.

ChrisHo2339 committed:
Add assertions

Add assertions to code and enabled assertions when using gradle.

ChrisHo1341 committed:
Merge pull request #16 from ChrisHo1341/branch-A-CodeQuality

Update code

ChrisHo1341 committed:
Merge branch 'master' into branch-A-Assertions

ChrisHo1341 committed:
Merge pull request #17 from ChrisHo1341/branch-A-Assertions

Add assertions

ChrisHo2339 committed:
Update TaskList.java

Update TaskList to handle situations where it contains duplicates.

TaskList.deleteDuplicate() will identify and delete duplicate tasks in the task list.

This method is called by typing "dupe" to the chatbot

ChrisHo1341 committed:
Merge pull request #18 from ChrisHo1341/branch-BCD-Extension

Update TaskList.java

ChrisHo2339 committed:
Add UserGuide

Add UserGuide

ChrisHo2339 committed:
Update Storage.java

Update Storage.java to fix issue where program would not work if directory was not found

ChrisHo2339 committed:
Update Ui.java

Update path to saved task list file.
